src/views/world/view_squad_spawner.cpp

```diff
@@ -0,0 +1,379 @@
+#include "services/gta_data/gta_data_service.hpp"
+#include "services/squad_spawner/squad_spawner.hpp"
+#include "views/view.hpp"
+
+namespace big
+{
+
+	void view::squad_spawner()
+	{
+		const char* const spawn_distance_modes[5]{"Custom", "On target", "Closeby", "Moderately distanced", "Far away"};
+		const char* const combat_ability_levels[3]{"Poor", "Average", "Professional"};
+
+
+		static player_ptr victim = g_player_service->get_selected();
+
+		static eSquadSpawnDistance spawn_distance_mode  = eSquadSpawnDistance::CLOSEBY;
+		static eCombatAbilityLevel combat_ability_level = eCombatAbilityLevel::AVERAGE;
+
+		static char name[32];
+		static char description[500];
+		static char ped_model[32];
+		static char veh_model[32];
+		static char weap_model[32];
+
+		static float ped_accuracy = 50.f;
+		static bool ped_invincibility;
+		static bool veh_invincibility;
+
+		static float ped_health;
+		static float ped_armor;
+
+		static float custom_spawn_distance;
+		static int squad_size = 1;
+		static bool ped_proofs[5];
+		static bool stay_in_veh                = false;
+		static bool spawn_behind_same_velocity = false;
+		static bool disperse                   = false;
```

Comment:
Move these values to the squad service/make a struct of these so you can easily reset them.

Reply:
Consider it done

Reply:
Wouldn't a static instance keep its ID and not increment since it would run the constructor only once? Right now I have switched to an instance but do spawn by creating another squad object to make sure the ID's are unique.